Raven.Database/Indexing/Index.cs

```diff
@@ -572,6 +572,12 @@ protected void Write(Func<RavenIndexWriter, Analyzer, IndexingWorkStats, Indexed
                                 }
                             }
                         }
+                        catch (OperationCanceledException)
+                        {
+                            //do not add error if this exception happens,
+                            //since this exception can happen during normal code-flow
+                            throw;
```

Comment:
This is different than in 3.0. Me and @arekpalinski do not think that 'eating' exception in this place is correct. If we do not want to add errors then it is ok, but indexing code should not proceed further. @ayende  and @myarichuk please verify our thinking. If we are correct then we need to fix 3.0 also.